File: client/src/types/inventoryRecord.ts
import { Product } from "./product"
import { Warehouse } from "./warehouse"

export interface InventoryRecord {
    id: number
    productId: number
    warehouseId: number
    quantity: number
    unit: string
    dateAdded: Date
    pricePerPiece: number

    product: Product
    warehouse: Warehouse
}

File: API/Models/InventoryRecord.cs
namespace API.Models
{
    public class InventoryRecord
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public int WarehouseId { get; set; }
        public int Quantity { get; set; }
        public string Unit { get; set; } = null!;
        public DateTime DateAdded { get; set; }
        public decimal PricePerPiece { get; set; }
    }
}
